packages/gestalt/src/InternalTextField.js

```diff
@@ -145,6 +151,10 @@ const InternalTextFieldWithForwardRef: React$AbstractComponent<Props, HTMLInputE
 
     const unstyledClasses = classnames(styles.unstyledTextField);
 
+    if (maxLength && maxLength?.characterCount < 0) {
```

Comment:
optional chaining is unnecessary after the preceding truthy check

Reply:
DONE